lib/utils/logDeprecation.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+const chalk = require('chalk');
+const memoizee = require('memoizee');
+
+const disabledCodesByEnv = extractCodes(process.env.SLS_DEPRECATION_DISABLE);
+
+const loggedDeprecations = new Set();
+
+function extractCodes(codesStr) {
+  if (!codesStr) {
+    return new Set();
+  }
+  return new Set(codesStr.split(','));
+}
+
+const resolveDeprecatedByService = memoizee((serviceConfig = {}) => {
+  let disabledDeprecations = [];
+  if (typeof serviceConfig.disabledDeprecations === 'string') {
+    disabledDeprecations = [serviceConfig.disabledDeprecations];
+  } else {
+    disabledDeprecations = serviceConfig.disabledDeprecations || [];
```

Comment:
To be on safer side, let's do `Array.from(serviceConfig.disabledDeprecations || [])`

Otherwise if e.g. a plain object is assigned to `disabledDeprecations`, it'll crash when it's passed to `new Set`, and we don't want crashes here

Reply:
I was planning to ask about that in a comment, but I think I didn't press the `comment` button.

Should we display a warning in case an object is passed for some reason? Or we ignore it for now until the schema validation is done?

Reply:
> Should we display a warning in case an object is passed for some reason? Or we ignore it for now until the schema validation is done?

Ignore it. It's also per JS convention to accept objects as empty arrays when array was expected.